### DISCUSSION TITLE
What makes a glb model interactable?

### DISCUSSION THREAD

**Discussion by hienvm** [sceneview/sceneview#441] [Q&A]
I want to play sound effects on touch, and also do some editting.
I tried this but to no avail. Or is it even possible? Am I misunderstanding?
![image](https://github.com/SceneView/sceneview-android/assets/102467543/38816295-48a5-47e6-a385-d5a4f9a2c59f)

**Comment by hienvm** [comment]
Ok, it turns out that the issue is on my glb models. I used the sample's one and it works just find.

That begs another question: What makes a model interactable?

I tried using blender to add collision / rigid body but that doesn't work.
Also tried to add a collision shape: modelNode.collisionShape = Box().

**Comment by hienvm** [comment]
Solved in 2.1.0.
Seems like an issue with the collision box.

**Comment by ThomasGorisse** [comment]
In v4.0.0-rc.1, any `ModelNode` is interactable by default. Just add gesture callbacks:

```kotlin
ModelNode(
    modelInstance = model,
    onTap = { event, node -> /* tapped */ },
    onDoubleTap = { event, node -> /* double-tapped */ },
    onLongPress = { event, node -> /* long-pressed */ }
)
```

The collision detection for tap targets uses the model's bounding box automatically. No special setup needed in the GLB — any model loaded with `rememberModelInstance` is tappable when you provide an `onTap` callback.
